### PR TITLE
Make the node to stop gracefully on startup error

### PR DESCRIPTION
**Pull request by diega**
Thread initialization is now outside constructors

### CODE REVIEW

**Comment by tinchou** [comment]
:four::zero::four: PR NOT FOUND

File: rskj-core/src/main/java/co/rsk/core/RskImpl.java [hunk 1]
```diff
@@ -29,12 +29,16 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
+@Component
 public class RskImpl extends EthereumImpl implements Rsk {
 
     private boolean isplaying;
     private final NodeBlockProcessor nodeBlockProcessor;
 
+    @Autowired
```
Comment (lsebrie) [inline rskj-core/src/main/java/co/rsk/core/RskImpl.java]: Is this autowired necessary?

File: rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java [hunk 2]
```diff
@@ -73,6 +73,7 @@
 
     private TxHandler txHandler;
 
+    @Autowired
```
Comment (lsebrie) [inline rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java]: same goes here, why would use @Autowired with @Component on same class?

File: rskj-core/src/main/java/co/rsk/core/RskImpl.java (hunk 1 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/core/RskImpl.java]: Yes, before this changes we needed to perform some initalization on `co.rsk.core.RskImpl` right after the instance was created and all this logic was part of `co.rsk.core.RskFactory`. Now all that complexity is gone and we just need to instantiate this class and inject the proper dependencies which is what accomplished here through the combination of *@Component* and *@Autowired* (in the constructor)

File: rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java (hunk 2 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java]: Same situation as in the previous comment, before this changes all was created and initialized in the `co.rsk.core.RskFactory`, now we would just need to create a new instance passing all the dependencies in the constructor and just return it, that's exactly what *@Component* and *@Autowired* do

**Comment by diega** [comment]
This doesn't pass the integration tests

**Pull request closed without merging**